File: androidApp/src/main/java/com/aitrainingapp/android/ui/login/LoginScreen.kt
package com.aitrainingapp.android.ui.login

import androidx.compose.animation.*
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.input.*
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.aitrainingapp.presentation.login.LoginEvent
import com.aitrainingapp.android.viewmodel.AndroidLoginViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun LoginScreen(
    viewModel: AndroidLoginViewModel,
    onLoginSuccess: () -> Unit
) {
    var username by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var isLoginMode by remember { mutableStateOf(true) }

    val state by viewModel.state.collectAsState()

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFF101010)) // ciemne tło siłowni
            .padding(24.dp)
    ) {
        Column(
            modifier = Modifier.align(Alignment.Center),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Text(
                text = if (isLoginMode) "Log in to AI Training App" else "Register New Account",
                style = MaterialTheme.typography.headlineMedium,
                color = Color.White
            )

            Spacer(modifier = Modifier.height(24.dp))

            OutlinedTextField(
                value = username,
                onValueChange = { username = it },
                label = { Text("Username") },
                singleLine = true,
                modifier = Modifier.fillMaxWidth(),
                colors = TextFieldDefaults.outlinedTextFieldColors(
                    focusedBorderColor = Color.Cyan,
                    unfocusedBorderColor = Color.Gray,
                    cursorColor = Color.Cyan,
                    focusedLabelColor = Color.Cyan,
                    unfocusedLabelColor = Color.LightGray,
                    containerColor = Color.Transparent,
                    disabledTextColor = Color.LightGray,
                    unfocusedTextColor = Color.White,
                    focusedTextColor = Color.White
                )
            )

            Spacer(modifier = Modifier.height(16.dp))

            OutlinedTextField(
                value = password,
                onValueChange = { password = it },
                label = { Text("Password") },
                singleLine = true,
                visualTransformation = PasswordVisualTransformation(),
                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                modifier = Modifier.fillMaxWidth(),
                colors = TextFieldDefaults.outlinedTextFieldColors(
                    focusedBorderColor = Color.Cyan,
                    unfocusedBorderColor = Color.Gray,
                    cursorColor = Color.Cyan,
                    focusedLabelColor = Color.Cyan,
                    unfocusedLabelColor = Color.LightGray,
                    containerColor = Color.Transparent,
                    disabledTextColor = Color.LightGray,
                    unfocusedTextColor = Color.White,
                    focusedTextColor = Color.White
                )
            )

            Spacer(modifier = Modifier.height(24.dp))

            Button(
                onClick = {
                    if (isLoginMode) {
                        viewModel.onEvent(LoginEvent.Login(username, password))
                    } else {
                        viewModel.onEvent(LoginEvent.Register(username, password))
                    }
                },
                modifier = Modifier.fillMaxWidth(),
                colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF00C853))
            ) {
                Text(if (isLoginMode) "LOGIN" else "REGISTER", fontSize = 18.sp)
            }

            Spacer(modifier = Modifier.height(12.dp))

            TextButton(onClick = { isLoginMode = !isLoginMode }) {
                Text(
                    text = if (isLoginMode) "No account? Register" else "Already registered? Login",
                    color = Color.LightGray
                )
            }

            if (state.isLoading) {
                Spacer(modifier = Modifier.height(16.dp))
                CircularProgressIndicator(color = Color.Cyan)
            }

            state.error?.let {
                Spacer(modifier = Modifier.height(16.dp))
                Box(
                    modifier = Modifier
                        .fillMaxWidth(),
                    contentAlignment = Alignment.Center
                ) {
                    Text(
                        text = "Błąd logowania: $it",
                        color = Color.Red,
                        style = MaterialTheme.typography.bodyMedium
                    )
                }
            }

            if (state.success) {
                LaunchedEffect(state.success) {
                    if (state.success) {
                        onLoginSuccess()
                    }
                }
            }
        }
    }
}

File: iosApp/ui/LoginScreen.swift
import SwiftUI

struct LoginScreen: View {
    @ObservedObject var viewModel: LoginViewModelWrapper
    var onLoginSuccess: () -> Void

    @State private var username = ""
    @State private var password = ""
    @State private var isLoginMode = true

    var body: some View {
        ZStack {
            Color(.black).ignoresSafeArea()

            VStack(spacing: 20) {
                Text(isLoginMode ? "Log in to AI Training App" : "Register New Account")
                    .font(.title)
                    .foregroundColor(.white)

                TextField("Username", text: $username)
                    .textFieldStyle(.roundedBorder)
                    .autocapitalization(.none)

                SecureField("Password", text: $password)
                    .textFieldStyle(.roundedBorder)

                Button(action: {
                    if isLoginMode {
                        viewModel.login(username: username, password: password)
                    } else {
                        viewModel.register(username: username, password: password)
                    }
                }) {
                    Text(isLoginMode ? "LOGIN" : "REGISTER")
                        .font(.headline)
                        .frame(maxWidth: .infinity)
                        .padding()
                        .background(Color.green)
                        .foregroundColor(.white)
                        .cornerRadius(10)
                }

                Button(action: {
                    isLoginMode.toggle()
                }) {
                    Text(isLoginMode ? "No account? Register" : "Already registered? Login")
                        .foregroundColor(.gray)
                        .font(.subheadline)
                }

                if viewModel.state.isLoading {
                    ProgressView().progressViewStyle(CircularProgressViewStyle(tint: .cyan))
                }

                if let error = viewModel.state.error {
                    Text("Błąd logowania: \(error)")
                        .foregroundColor(.red)
                        .font(.subheadline)
                }
            }
            .padding(24)
        }
        .onChange(of: viewModel.state.success) { success in
            if success {
                onLoginSuccess()
            }
        }
    }
}
